rstudio/rstudio
# automatically activate .condaenv environments; fixes for reticulate binding to Anaconda

### Intent

This PR contains two main changes:

1. If the project contains a Conda environment within the `.condaenv` directory, RStudio will now automatically activate that environment on startup.

2. Add some extra glue to ensure that `reticulate` can bind to Anaconda installations on macOS. See https://github.com/ContinuumIO/anaconda-issues/issues/12478 for more details.

### Approach

The Anaconda glue is the more interesting part of the PR. Currently, Anaconda installations rely on the process rpath when discovering libraries. This unfortunately doesn't work for us, since we attempt to load Python as a library (and so don't have the rpath entries set that Anaconda is expecting).

The workaround is to set `DYLD_FALLBACK_LIBRARY_PATH` to include the Anaconda library path, but this is challenging for a couple reasons:

1. The `DYLD_FALLBACK_LIBRARY_PATH` cannot be changed after a process has already been launched. (You can change the value of the environment variable, but it will not affect future library load attempts in the current process.)
2. When RStudio launches the `rsession` process, we have no idea if the user plans to use Python, or even Anaconda Python. We also don't know where Anaconda might be installed.

The solution is to instead have RStudio place a special fallback library path in the `DYLD_FALLBACK_LIBRARY_PATH`, and have R initialize the library path when necessary. See https://github.com/rstudio/reticulate/commit/08f145b0d2f13c8f8c242ea788359406d139467d for an example.

### Automated Tests

N/A

### QA Notes

Verify, on macOS, that the following works:

1. Install Anaconda: https://www.anaconda.com/products/individual#macos
2. Install the development version of `reticulate`: `remotes::install_github("rstudio/reticulate")`
3. Try to load that environment: `reticulate::use_python("~/opt/anaconda3/bin/python", required = TRUE); reticulate::py_config()`

If all is well, you should see `reticulate::py_config()` print without issue.

### Checklist
